### PR TITLE
    CRM-20670 fix editing membership types with a lot of members

### PR DESCRIPTION
**Pull request by JO0st**
- fixes the editing of membership types with a lot of members
[https://issues.civicrm.org/jira/browse/CRM-20670](https://issues.civicrm.org/jira/browse/CRM-20670)

### CODE REVIEW

**Comment by seamuslee001** [comment]
Jenkins ok to test

**Comment by seamuslee001** [comment]
@eileenmcnaughton @totten can one of you get Jenkins to run on this pls

**Review by seamuslee001 — APPROVED** [review]
Changes look ok, just suggesting @JO0st you could also remove L190 and L191 here as they wouldn't be neeeded as i don't believe $membershipType is used in the rest of the buildQuickForm function

File: CRM/Member/Form/MembershipType.php [hunk 1]
```diff
@@ -187,12 +187,11 @@ public function buildQuickForm() {
...
-      $membershipType = new CRM_Member_BAO_Membership();
-      $membershipType->membership_type_id = $this->_id;
-      if ($membershipType->find(TRUE)) {
-        $membershipRecords = TRUE;
-        $memberRel->freeze();
-      }
+        $result = civicrm_api3("Membership", "get", ["membership_type_id"=>$this->_id, "options"=>["limit"=>1]]);
+        $membershipRecords = $result["count"] > 0;
+        if ($membershipRecords) {
+            $memberRel->freeze();
```
Comment (seamuslee001) [inline CRM/Member/Form/MembershipType.php]: @JO0st 2 things this you will want to still assign $membershipRecords as it gets passed to the template at L197 and also CiviCRM style checker won't pass the array sort version need it in the long format array()

Comment (JO0st) [inline CRM/Member/Form/MembershipType.php]: MembershipRecords still gets set, only a couple of lines earlier (L191). I added a commit to change the array style.

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
I've set off testing. I agree this makes sense although I think this

$membershipRecords = $result["count"] > 0;

would be more readable as 

$membershipRecords = ($result["count"] > 0);

Or you could use the count key -ie

if ($membershipRecords['count']) {

}

'count' is always populated when you use get on an api so this is enotice-safe.

Do you know how to squash your commits into a single commit? It's nice to tidy them up into one

**Comment by eileenmcnaughton** [comment]
Let us know if you have any trouble interpreting the Jenkins style warnings - https://test.civicrm.org/job/CiviCRM-Core-PR/15688/checkstyleResult/new/

We use drupal coding standards https://www.drupal.org/docs/develop/standards/coding-standards

**Comment by JO0st** [comment]
I've fixed the jenkins style warnings, but can't find a way to squash the commits in a way I can push it back to github.

**Comment by eileenmcnaughton** [comment]
So I normally do this

``
git pull --rebase upstream/master
git fetch -f
git rebase -i upstream/master
git push -f mybranch master
``
in the rebase -i stage you need to mark the commits to be squashed with 's' to squash them

**Comment by seamuslee001** [comment]
@JO0st Joost on my naked eye it looks like everything is indented one too few spaces

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by JO0st** [comment]
I think I have fixed the styling issues. I also squashed everything back into one commit. 
Sorry for the stupid mistakes in the first styling fix.

**Review by seamuslee001 — APPROVED** [review]
This makes sense to me and looks good

**Comment by totten** [comment]
jenkins, ok to test

**Comment by seamuslee001** [comment]
@JO0st one last style issue https://test.civicrm.org/job/CiviCRM-Core-PR/15828/checkstyleResult/new/

**Comment by seamuslee001** [comment]
@eileenmcnaughton @totten can you get Jenkins running again?

**Comment by seamuslee001** [comment]
or @totten fix the space (as maintainer you should be able to edit)

**Comment by JO0st** [comment]
I have removed the space. Is there any way I can run the style checks on my local machine?
That way I can avoid having to push that many times for style issues in the future.

**Comment by seamuslee001** [comment]
@JO0st if you have buildkit installed there is a tool civilint which will run things through the style checker for you its a CLI tool

**Comment by seamuslee001** [comment]
@eileenmcnaughton @totten this is passed tests now

**Comment by eileenmcnaughton** [comment]
merging per advice from @seamuslee001 - would have preferred brackets around 

$membershipRecords = ($result["count"] > 0);

but it should be fine without

**Pull request merged**